dashboard/src/actions/overviewActions.js

```diff
@@ -0,0 +1,189 @@
+        const newRuns = data.filter(
+          (item) => !item.metadata["dashboard.saved"]
+        );
+
+        dispatch({
+          type: TYPES.SAVED_RUNS,
+          payload: savedRuns,
+        });
+        dispatch({
+          type: TYPES.NEW_RUNS,
+          payload: newRuns,
+        });
+        dispatch({
+          type: TYPES.INIT_NEW_RUNS,
+          payload: newRuns?.slice(0, defaultPerPage),
+        });
+      }
+    }
+  } catch (error) {
+    dispatch(constructToast("danger", error?.response?.data?.message));
+    dispatch({ type: TYPES.NETWORK_ERROR });
+  }
+  dispatch({ type: TYPES.COMPLETED });
+};
+
+export const updateDataset =
+  (dataset, actionType, updateAction) => async (dispatch, getState) => {
+    try {
+      const actions = {
+        save: "dashboard.saved",
+        read: "dashboard.seen",
+        favorite: "user.favorite",
+      };
+      dispatch({ type: TYPES.LOADING });
+      const savedRuns = getState().overview.savedRuns;
+      const newRuns = getState().overview.newRuns;
+      const initNewRuns = getState().overview.initNewRuns;
+
+      const method = actions[actionType];
+
+      const dataIndex = newRuns.findIndex(
+        (item) => item.resource_id === dataset.resource_id
+      );
+      const endpoints = getState().apiEndpoint.endpoints;
+      const response = await API.put(
+        `${endpoints?.api?.datasets_metadata}/${dataset.resource_id}`,
+        {
+          metadata: { [method]: updateAction },
+        }
+      );
+      if (response.status === 200) {
+        if (actionType === "save") {
+          savedRuns.push(dataset);
+          const filteredNewRuns = newRuns.filter(
+            (item) => item.resource_id !== dataset.resource_id
+          );
+          const filteredInitRuns = initNewRuns.filter(
+            (item) => item.resource_id !== dataset.resource_id
+          );
+          dispatch({
+            type: TYPES.SAVED_RUNS,
+            payload: savedRuns,
+          });
+          dispatch({
+            type: TYPES.INIT_NEW_RUNS,
+            payload: filteredInitRuns,
+          });
+          dispatch({
+            type: TYPES.NEW_RUNS,
+            payload: filteredNewRuns,
+          });
+        } else {
+          newRuns[dataIndex].metadata[actions[actionType]] =
+            response.data[actions[actionType]];
+          dispatch({
+            type: TYPES.NEW_RUNS,
+            payload: newRuns,
+          });
+        }
+      } else {
+        dispatch(constructToast("danger", response?.data?.message));
+      }
+    } catch (error) {
+      dispatch(constructToast("danger", error?.response?.data?.message));
+      dispatch({ type: TYPES.NETWORK_ERROR });
+    }
+    dispatch({ type: TYPES.COMPLETED });
+  };
+
+export const deleteDataset = (dataset) => async (dispatch, getState) => {
+  try {
+    dispatch({ type: TYPES.LOADING });
+    const endpoints = getState().apiEndpoint.endpoints;
+    const datasets = getState().overview.newRuns;
```

Comment:
Since this assignment is only used at line 137, you might want to consider moving it into that block.

Reply:
Moved